opa-filter-core/src/main/java/com/bisnode/opa/spring/security/filter/Jwt.java

```diff
@@ -32,13 +34,22 @@ private Jwt(String encodedToken) {
             return Optional.empty();
         }
 
-        if (!(authentication.getCredentials() instanceof AbstractOAuth2Token)) {
-            log.warn("Authentication credentials not an instance of AbstractOAuth2Token.");
-            return Optional.empty();
+        // For Spring Security in Spring < 5
+        if (authentication.getDetails() instanceof OAuth2AuthenticationDetails) {
+            final String accessToken = ((OAuth2AuthenticationDetails) authentication.getDetails()).getTokenValue();
+            if (nonNull(accessToken) && !accessToken.isBlank()) {
+                return Optional.of(accessToken);
+            }
+        }
+
+        // For Spring Security in Spring >= 5
+        if (authentication.getCredentials() instanceof AbstractOAuth2Token) {
```

Comment:
Should this not be an `else if` since the block above is guaranteed to not happen if this is true?

Reply:
It depends on client's config, there are no guarantees.